lib/puppet/indirector/request.rb

```diff
@@ -250,8 +250,6 @@ def set_uri_key(key)
       return
     end
 
-    env, indirector, @key = URI.unescape(uri.path.sub(/^\//, '')).split('/',3)
-    @key ||= ''
-    self.environment = env unless env == ''
+    @key = key
```

Comment:
Also worth noting that this code is not only used by faces. The agent will use it as well when `file` resources are used with the `filebucket` type.